### ISSUE TITLE
Improve RequestStack shim rules

### ISSUE THREAD

**Issue by mglaman** [mglaman/phpstan-drupal#458]
DrupalRequestStackShimInClassMethodRule and DrupalRequestStackShimInstantiationRule were added to detect usages of `Symfony\Component\HttpFoundation\RequestStack`. Jumping from Symfony 4 -> 6 doesn't catch deprecated calls to `getMasterRequest` which was renamed to `getMainRequest`.

Instead of telling users to fix their namespace in `use`, maybe we should just provide the same deprecations as the shim does, so only valid calls need to be fixed?

From \Drupal\Core\Http\RequestStack

```
  public function getMainRequest(): ?Request {
    if (method_exists(SymfonyRequestStack::class, 'getMainRequest')) {
      return parent::getMainRequest();
    }
    else {
      return parent::getMasterRequest();
    }
  }
```

```
  public function getMasterRequest() {
    @trigger_error(__METHOD__ . '() is deprecated in drupal:9.3.0 and is removed from drupal:10.0.0. Use getMainRequest() instead. See https://www.drupal.org/node/3253744', E_USER_DEPRECATED);
    return $this->getMainRequest();
  }
```

This could be less noisy and avoid changing code which only runs `getCurrentRequest`, the most common use case.
